contracts/vesting/src/vesting.rs

```diff
@@ -0,0 +1,559 @@
+        let created_account_test = get_account(&deps.storage, &Addr::unchecked("fixture"));
+        assert_eq!(Some(&account), created_account.as_ref());
+        assert_eq!(Some(&account), created_account_test.as_ref());
+        assert_eq!(
+            get_account_balance(&deps.storage, &account.address),
+            Some(Uint128(1_000_000_000_000))
+        );
+        assert_eq!(
+            account.get_balance(&deps.storage),
+            Uint128(1_000_000_000_000)
+        )
+    }
+
+    #[test]
+    fn test_period_logic() {
+        let mut deps = init_contract();
+        let env = mock_env();
+
+        let account = vesting_account_fixture(&mut deps.storage, &env);
+
+        assert_eq!(account.periods.len(), NUM_VESTING_PERIODS as usize);
+        assert_eq!(account.periods.len(), 8);
+
+        let current_period = account.get_current_vesting_period(Timestamp::from_seconds(0));
+        let next_period = account.get_next_vesting_period(Timestamp::from_seconds(0));
+        assert_eq!(0, current_period);
+        assert_eq!(0, next_period);
+
+        let block_time = Timestamp::from_seconds(account.start_time.seconds() + VESTING_PERIOD + 1);
+        let current_period = account.get_current_vesting_period(block_time);
+        let next_period = account.get_next_vesting_period(block_time);
+        assert_eq!(current_period, 1);
+        assert_eq!(next_period, 2);
+        let vested_coins = account.get_vested_coins(Some(block_time), &env);
+        let vesting_coins = account.get_vesting_coins(Some(block_time), &env);
+        assert_eq!(
+            vested_coins.amount,
+            Uint128(account.get_original_vesting().amount.u128() / NUM_VESTING_PERIODS as u128)
+        );
+        assert_eq!(
+            vesting_coins.amount,
+            Uint128(
+                account.get_original_vesting().amount.u128()
+                    - account.get_original_vesting().amount.u128() / NUM_VESTING_PERIODS as u128
+            )
+        );
+
+        let block_time =
+            Timestamp::from_seconds(account.start_time.seconds() + 5 * VESTING_PERIOD + 1);
+        let current_period = account.get_current_vesting_period(block_time);
+        let next_period = account.get_next_vesting_period(block_time);
+        assert_eq!(current_period, 5);
+        assert_eq!(next_period, 6);
+        let vested_coins = account.get_vested_coins(Some(block_time), &env);
+        let vesting_coins = account.get_vesting_coins(Some(block_time), &env);
+        assert_eq!(
+            vested_coins.amount,
+            Uint128(5 * account.get_original_vesting().amount.u128() / NUM_VESTING_PERIODS as u128)
+        );
+        assert_eq!(
+            vesting_coins.amount,
+            Uint128(
+                account.get_original_vesting().amount.u128()
+                    - 5 * account.get_original_vesting().amount.u128()
+                        / NUM_VESTING_PERIODS as u128
+            )
+        );
```

Comment:
It feels like it could be useful to have one additional case here of when we reach the final period (or maybe also for one before it)